### ISSUE TITLE
Error after upgrading to v2.16.1

### ISSUE THREAD

**Issue by kbond** [FriendsOfPHP/PHP-CS-Fixer#4684]
I get the following error after upgrading to 2.16.1 (from 2.15.0):

```
PHP Fatal error:  Uncaught TypeError: Argument 2 passed to Symfony\Component\EventDispatcher\EventDispatcher::dispatch() must be an instance of Symfony\Component\EventDispatcher\Event or null, instance of PhpCsFixer\FixerFileProcessedEvent given, called in /vendor/friendsofphp/php-cs-fixer/src/Runner/Runner.php on line 300 and defined in /vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/EventDispatcher.php:37
```

#### The PHP version you are using (`$ php -v`):
=> 7.2.25

#### PHP CS Fixer version you are using (`$ php-cs-fixer -V`):
=> 2.16.1

#### The command you use to run PHP CS Fixer:
=> vendor/bin/php-cs-fixer fix -v

#### The configuration file you are using, if any:
```php
$finder = PhpCsFixer\Finder::create()
    ->exclude('var')
    ->exclude('node_modules')
    ->exclude('etc')
    ->exclude('build')
    ->exclude('vendor')
    ->exclude('web')
    ->in(__DIR__)
;

return PhpCsFixer\Config::create()
    ->setRules([
        '@Symfony' => true,
        '@Symfony:risky' => true,
        '@DoctrineAnnotation' => true,
        '@PHP71Migration' => true,
        '@PHP71Migration:risky' => true,
        '@PHPUnit75Migration:risky' => true,
        'multiline_whitespace_before_semicolons' => ['strategy' => 'new_line_for_chained_calls'],
        'multiline_comment_opening_closing' => true,
        'array_syntax' => ['syntax' => 'short'],
        'ordered_imports' => true,
        'ordered_class_elements' => true,
        'native_function_invocation' => true,
        'explicit_indirect_variable' => true,
        'explicit_string_variable' => true,
        'escape_implicit_backslashes' => true,
        'mb_str_functions' => true,
        'logical_operators' => true,
        'php_unit_method_casing' => ['case' => 'snake_case'],
        'php_unit_test_annotation' => ['style' => 'annotation'],
        'no_unreachable_default_argument_value' => true,
        'declare_strict_types' => false,
        'void_return' => false,
        'single_trait_insert_per_statement' => false,
        'simple_to_complex_string_variable' => true,
        'no_superfluous_phpdoc_tags' => ['allow_mixed' => true],
    ])
    ->setRiskyAllowed(true)
    ->setUsingCache(true)
    ->setFinder($finder)
;
```

**Comment by keradus** [comment]
hey @kbond !
I have problems while trying to reproduce this issue.
Please, can you share result of `composer info`?
Even better - maybe make a small github repo that we can replicate the issue on ?

**Comment by kbond** [comment]
I was using this as a project dependency on a Symfony 3.4 app. I see this is no longer (maybe it never was) recommended as a way to install this package. I have installed globally as described in the docs and no longer have this issue.

If you'd still like to diagnose this issue, I can create a reproducer, otherwise, we can close this issue.

Let me know!

**Comment by kbond** [comment]
Ah, pretty sure I found the issue: I'm using a Symfony 3.4 app but I also pull in the `HttpClient` from 4.x. This pulls in the `symfony/contracts` package [this](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/9ac66c143b35c02fb60563b97d1b127a6e51217d/src/Event/Event.php#L18) evaluates to `true` and causes the issue.

**Comment by keradus** [comment]
thanks for update, @kbond !
actually, if you would still be able to provide repo where I can reproduce, i would take a deeper look.

agree, this is edge case, still sth we can try to fix

**Comment by keradus** [comment]
@kbond , I believe I reproduced. can you look at #4685 please ?

**Comment by 0mars** [comment]
issue still exists in 2.16

**Comment by SpacePossum** [comment]
hi @0mars ,

did you try the latest of the 2.16 branch or the latest 2.16 release, as the bug fix @keradus patched on the 2.16 branch has not been released yet (will in 2.16.5)

**Comment by 0mars** [comment]
@SpacePossum last one is https://packagist.org/packages/friendsofphp/php-cs-fixer#v2.16.1 on packagist, I upgraded to dev branch and it works
